src/bsoncxx/types/value.cpp

```diff
@@ -40,15 +45,28 @@ value::value(const value& rhs) {
 }
 
 value& value::operator=(const value& rhs) {
-    _type = rhs._type;
+    if(_type && !rhs._type) {
```

Comment:
Yes, assigning to a data member of a union makes the other data member unassigned. In that case, I think it might make sense to leave this as is, since then the block that clears out the union won't run in the case of self-assignment.